### DISCUSSION TITLE
Can I use a wallet client to read from a contract?

### DISCUSSION THREAD

**Discussion by richardryangarcia** [wevm/viem#1703] [General]
I noticed that when creating a contract instance with a wallet client, the `read` function is not accessible. Its only when you use a public client that you have access to read. 
Similarly, when trying to use the clients directly, public clients have the `readContract` option while wallet clients do not. 

I would like to read a contract with the caller address in scope (not the public client). Is there a way to do this using Viem? I feel like Ive seen that working before but now digging through the docs (and types) it doesnt seem to be possible but hopefully Im just missing something.

Thanks for the help yall!

**Comment by jxom** [comment]
Check here: https://viem.sh/docs/clients/wallet#optional-extend-with-public-actions

**Reply by jxom** [reply]
can use with a contract instance like this:

```tsx
const client = createWalletClient({
  account,
  chain: mainnet,
  transport: http()
}).extend(publicActions) 

const contract = getContract({ abi, address, client })
```

**Reply by richardryangarcia** [reply]
Excellent! Thank you for the quick response 🙏

**Reply by richardryangarcia** [reply]
Does this only work for certain chains? I tried switching to Base Sepolia and now there seems to be a type error with publicActions. It works fine with mainnet. 

here is my client config 

`const client = createWalletClient({
  account: privateKeyToAccount(
    "<PRIVATE_KEY>"
  ),
  chain: baseSepolia,
  transport: http(
    "https://base-sepolia.g.alchemy.com/v2/<ALCHEMY_API_KEY>"
  ),
}).extend(publicActions);
`

**Reply by jxom** [reply]
Do you mind breaking this minimal reproducible example?

https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAbzgYygUwIYzQdQwG3zRgGF9g0A7GAGjgAsYYw6wBXAI3OQEFkZgESgGc4AXzgAzKBBBwA5ADcKIeQChQkWIjgcMwtAGU0kchnFSZcpSoD0yehmAj1a5EOHxk5KvAC8KOhYuAREpD7UABRIanBxcBjI7mzUAFwKAAwAHvJ0sfEOTpTpegbGpsAYefFwMFAYIlow6YzMkXAARK1gwqm2tqVoALQGFRgAdADm4wQOaCAAnuPuILaKAEy2ADw8ADIkABIAogCyAJoA+jwACgCSFwDSR2cAfB1wAJRqYh-jaFnYSgAE0i7C4wF4-EEIg+QA

Are you on the latest version, etc?

**Reply by richardryangarcia** [reply]
ahh yes upgrading the version resolved it. Didnt realize I was outdated. Thanks again for the help!
